openedx/core/lib/xblock_keys/learning_context_key.py

```diff
@@ -0,0 +1,56 @@
+from opaque_keys import OpaqueKey, InvalidKeyError
+
+from .usage_locator import GlobalUsageLocator
+
+class LearningContextKey(OpaqueKey):
+    """
+    A key that idenitifies a course, a library, a program,
+    or some other collection of content where learning happens.
+    """
+    KEY_TYPE = 'context_key'
+    __slots__ = ()
+
+    def make_usage_key(self, definition_key, usage_id=None):
+        """
+        Return a usage key, given the given the specified definition key and usage_id
+        This function should not actually create any new ids, but should simply
+        return one that already exists.
+        """
+        raise NotImplementedError()
+
+
+class GlobalContextLocator(LearningContextKey):
```

Comment:
Why an explicitly separate class for this rather than making `global` just another named instance of a `LearningContextKey`?

Reply:
If I understand you correctly, this is a separate class mainly so that it can implement `make_usage_key(definition_id)`, which is an abstract method in `LearningContextKey`.